internal/server/data/data.go

```diff
@@ -152,28 +139,68 @@ func add[T models.Modelable](db *gorm.DB, model *T) error {
 		return err
 	}
 
-	if err := db.Create(model).Error; err != nil {
-		if isUniqueConstraintViolation(err) {
-			return fmt.Errorf("%w: %s", internal.ErrDuplicate, err)
-		}
+	err := db.Create(model).Error
+	return handleError(err)
+}
 
-		return err
-	}
+type UniqueConstraintError struct {
+	Table  string
+	Column string
+}
 
-	return nil
+func (e UniqueConstraintError) Error() string {
+	return fmt.Sprintf("value for %v already exists in %v", e.Column, e.Table)
```

Comment:
We should probably have a check here for `if e.Column == ""` in the case where we failed to parse the message, so that we can provide a fallback message that reads better than "value for already exists in"